src/Microsoft.AspNetCore.Server.Kestrel/Filter/StreamSocketOutput.cs

```diff
@@ -65,14 +70,39 @@ public void ProducingComplete(MemoryPoolIterator end)
             var block = _producingBlock;
             while (block != end.Block)
             {
-                _outputStream.Write(block.Data.Array, block.Data.Offset, block.Data.Count);
+                // Writes can fail if the client has closed the connection but we're still sending a response.
+                // If we don't handle an exception from _outputStream.Write() here, we'll leak memory blocks.
+                if (_canWrite)
+                {
+                    try
+                    {
+                         _outputStream.Write(block.Data.Array, block.Data.Offset, block.Data.Count);
+                    }
+                    catch (Exception e)
```

Comment:
Just to confirm: do we want to suppress all types of exceptions and prevent future write?
The logged msg may not be accurate as the exceptions according to API could be:
        // Exceptions:
        //   T:System.ArgumentException:
        //     The sum of offset and count is greater than the buffer length.
        //
        //   T:System.ArgumentNullException:
        //     buffer is null.
        //
        //   T:System.ArgumentOutOfRangeException:
        //     offset or count is negative.
        //
        //   T:System.IO.IOException:
        //     An I/O error occured, such as the specified file cannot be found.
        //
        //   T:System.NotSupportedException:
        //     The stream does not support writing.
        //
        //   T:System.ObjectDisposedException:
        //     System.IO.Stream.Write(System.Byte[],System.Int32,System.Int32) was called after
        //     the stream was closed.